meta-oe/recipes-devtools/perfetto/perfetto.bb

```diff
@@ -0,0 +1,145 @@
+                    file://buildtools/libcxxabi/LICENSE.TXT;md5=7b9334635b542c56868400a46b272b1e \
+                    file://buildtools/libunwind/LICENSE.TXT;md5=f66970035d12f196030658b11725e1a1 \
+                    file://buildtools/linenoise/LICENSE;md5=faa55ac8cbebebcb4a84fe1ea2879578 \
+                    file://buildtools/lzma/Java/Tukaani/COPYING;md5=3d9c8c11183717c7a0539a9afccec6e4 \
+                    file://buildtools/protobuf/LICENSE;md5=37b5762e07f0af8c74ce80a8bda4266b \
+                    file://buildtools/sqlite_src/LICENSE.md;md5=82c5c312251c7bb3f775273ef25592e2 \
+                    file://buildtools/sqlite_src/autoconf/tea/license.terms;md5=5e0405ae78edb434653628790bb02b17 \
+                    file://buildtools/zlib/LICENSE;md5=f09575dbfb09420642318b413159496f \
+                    file://debian/copyright;md5=4e08364c82141f181de69d0a2b89d612 \
+                    file://python/LICENSE;md5=c602a632c34ade9c78a976734077bce7"
+
+SRC_URI = "git://github.com/google/perfetto.git;protocol=https"
+SRC_URI[sha256sum] = "3d92dc8f561ea6a83f3efdea33c5aeefa4a3cd98f9739190efd9c33c30bcbdf0"
+SRCREV = "5bd3f582c075d0d026c5fe0b5e291d34dee0d976"
+
+DEPENDS += " ninja-native"
+DEPENDS += " curl-native"
+DEPENDS += " ca-certificates-native"
+
+# To avoid the GNU_HASH warning
+TARGET_CC_ARCH += "${LDFLAGS}"
+
+S = "${WORKDIR}/git"
+
+FILES:${PN}:append = " \
+  ${bindir}/tracebox \
+  "
+
+# Perfetto includes an "install-build-deps" script. It requires network connection.
+do_fetch () {
+    # Fixme: This PATH is for making curl visible to bitbake, there should be a better way
+    export CURL_CA_BUNDLE=`dirname dirname ${PSEUDO_PREFIX}`"/ca-certificates-native/etc/ssl/certs/ca-certificates.crt"
+    export PATH=$PATH":/usr/bin"
+    
+    if [ ! -d perfetto -a ! -d perfetto.git ]; then
+        # FIXME: Make this work with base_do_fetch
+        HTTPS_SRCURI=`echo "${SRC_URI}"|sed "s/git\:\/\//https:\/\//g;s/;protocol=.*//g"`
+        
+        # The bare Git repository for copying into git2
+        git clone --bare $HTTPS_SRCURI
+        
+        if [ -f git2/github.com.google.perfetto.git.lock ]; then
+            rm git2/github.com.google.perfetto.git.lock
+        fi
+        if [ -d git2/github.com.google.perfetto.git ]; then
+            rm -rf git2/github.com.google.perfetto.git
+        fi
+        
+        mv perfetto.git git2/github.com.google.perfetto.git
+        touch git2/github.com.google.perfetto.git.done
+        
+        # The ordinary Git repository
+        git clone $HTTPS_SRCURI
+
+        cd perfetto
+        git reset --hard ${SRCREV}
+        echo "Running tools/install-build-deps script"
+        python3 tools/install-build-deps
+        cd ..
+        tar cfz perfetto.tar.gz perfetto
+        touch perfetto.tar.gz.done
+        rm -r -f perfetto
+    fi
+}
+
+do_unpack() {
+    tar xfz ../../../../../downloads/perfetto.tar.gz
```

Comment:
This won't work for with any modification to DL_DIR or TMPDIR, at least use DL_DIR but both do_fetch and do_unpack should be simplified as mentioned above.

Reply:
Thanks for the suggestion, I replaced `../../../../../downloads` with DL_DIR.

Is the "simplified" referring to "express the SRC_URIs to include all thats needed"? I'm trying to figure out how to do it.

Reply:
This is not relevant now because the manual unpack step is eliminated.